src/App.js

```diff
@@ -1,85 +1,200 @@
+import { Filters } from './components/Filters/Filters';
+
+class App extends Component {
+  state = {
+    filtersFlag: 1,
+    count: 0,
+    rawTodos: [],
+    visibleTodos: [],
+    todoLastId: 0,
+  }
+
+  componentDidMount() {
+    const todos = localStorage.todos
+      ? JSON.parse(localStorage.getItem('todos'))
+      : [];
+    const lastId = localStorage.todoLastId
+      ? +localStorage.getItem('todoLastId')
+      : 0;
+
+    this.setState({
+      count: todos.reduce((acc, todo) => (todo.completed ? acc : acc + 1), 0),
+      rawTodos: todos,
+      visibleTodos: todos,
+      todoLastId: lastId,
+    });
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('todos', JSON.stringify(this.state.rawTodos));
+    localStorage.setItem('todoLastId', this.state.todoLastId);
+  }
+
+  checkBoxHandler = (id) => {
+    const todos = [...this.state.rawTodos];
+    const currentTodo = todos.find(todo => todo.id === id);
+
+    currentTodo.completed = !currentTodo.completed;
+
+    this.setState(prevState => ({
+      rawTodos: [...todos],
+      visibleTodos: this.filter(prevState.filtersFlag, todos),
+      count: todos.reduce((acc, item) => (item.completed ? acc : acc + 1), 0),
+    }));
+  }
+
+  onClearButton = () => {
+    const todos = this.state.rawTodos.filter(todo => !todo.completed);
+
+    this.setState(prevState => ({
+      rawTodos: [...todos],
+      visibleTodos: this.filter(prevState.filtersFlag, todos),
+      count: todos.length,
+    }));
+  }
+
+  destroyHandler = (id) => {
+    const todos = this.state.rawTodos.filter(todo => todo.id !== id);
+
+    this.setState(prevState => ({
+      rawTodos: [...todos],
+      count: todos.reduce((acc, item) => (item.completed ? acc : acc + 1), 0),
+      visibleTodos: this.filter(prevState.filtersFlag, todos),
+    }));
+  }
+
+  editHandler = (id, title) => {
+    const todos = [...this.state.rawTodos];
+
+    (todos.find(item => item.id === id)).title = title;
+    this.setState(prevState => ({
+      rawTodos: [...todos],
+      visibleTodos: this.filter(prevState.filtersFlag, todos),
+    }));
+  }
+
+  newTodoHandler = (title) => {
+    const todos = [...this.state.rawTodos];
+    const newId = this.state.todoLastId + 1;
+
+    todos.push({
+      id: newId,
+      title,
+      completed: false,
+    });
 
-      <footer className="footer">
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button type="button" className="clear-completed">
-          Clear completed
-        </button>
-      </footer>
-    </section>
-  );
+    this.setState(prevState => ({
+      count: todos.length,
+      rawTodos: [...todos],
```

Comment:
you've already copied it

Reply:
Я скопировал со стейта масим внес в него изменения и в стейт деструктурирую измененый масив

Reply:
first of all, it's not a destructuring, it's called spread operator
and in addition, you don't need to do it since you copied an array from the state on 81st line